src/flash_loader.c

```diff
@@ -262,6 +262,7 @@ int stlink_flash_loader_write_to_sram(stlink_t *sl, stm32_addr_t* addr, size_t*
         loader_code = loader_code_stm32l;
         loader_size = sizeof(loader_code_stm32l);
     } else if (sl->core_id == STM32VL_CORE_ID
+            || sl->chip_id == CS32VL_CORE_ID
```

Comment:
You are comparing Chip ID with Core ID.
Even if you fix it and will compares sl->core_id then this code will break work with many STM32F4 chips with Core ID 0x2ba01477
See #757 and #761 

What Chip ID printed if you run "st-info --probe"?

Reply:
Oh my bad!
I feel ashamed.
chipid is 0x0410 and coreid is 0x2ba01477

Reply:
0x410 (STLINK_CHIPID_STM32_F1_MEDIUM) is already in use and should work. Please check latest version. CS32F103 support ( #805 ) was recently added (3 month ago).

Reply:
The problem is **not** with the chipid. It's the coreid that is unknown.
For this chipid it is waiting for 0x1ba01477 not 0x2ba01477
That's why it says: `ERROR flash_loader.c: unknown coreid, not sure what flash loader to use, aborting! coreid: 2ba01477, chipid: 410` see issue #833
It's not a real STM32F103C8T6, it's a CKS32F103C8T6

Reply:
Message about unknown coreid is not entirely correct. Function stlink_flash_loader_write_to_sram() uses chipid to select flash loader type. And only for all STM32F1 and STM32F7 chips function uses coreid. If function can not detect chip, it prints a message. I think message must be "unknown chipid and coreid" or "can not detect chip type".

CKS32F103C8T6 and STM32F103C8T6 have same chipid, but CKS32 uses incorrect coreid. In #805 added check for STLINK_CHIPID_STM32_F1_MEDIUM and now CS/CKS32F103C8T6 should works. Did you try latest version?

Reply:
I don't understand.
I've just downloaded latest version, compiled it, and it works fine.
I was sure I was using last version. Maybe a PATH issue with multiple versions.
This PR becomes useless.
Sorry for the wasted time.

Reply:
Issue featured in Release v1.6.1 which resolves this discussion.